src/SdkCommon/ClientRuntime/ClientRuntime/ExtensibleEnum.cs

```diff
@@ -119,16 +119,6 @@ public override string ToString()
             return allKnownValues.ToDictionary(v => v._value, v => v);
         }
 
-        public static T Create(V value)
-        {
-            var existingItem = Lookup(value);
-            if (existingItem!=null)
-            {
-                return existingItem;
-            }
-            var newItem = (T)Activator.CreateInstance(typeof(T), new object[] { value });
-            _valueMap.TryAdd(value, newItem);
-            return newItem;
-        }
+        protected static void CacheValue(V value, T newItem) => _valueMap.TryAdd(value, newItem);
```

Comment:
My objection to modifying the map still stands. Not worth the effort relative to the cognitive load, runtime overhead, and chance of bugs that it introduces.

Reply:
I'm still unsure as to why caching newly introduced values is a bad idea. I agree it may be a bit of an overhead but I also feel it helps keep track of values created of that type (though I can't think of a concrete scenario where this is useful). I'm on the fence regarding this.

@olydis @fearthecowboy feedback would be much appreciated

Reply:
@brjohnstmsft what was the original motivation of the `Dictionary` in your case? I assumed it is to get referential equality as you would get it from C# enums.

E.g., when the server returns `{ "color": "red" }`, I wanna be able to say `response.Color == Color.Red` after deserialization. Then also, the extensibility of extensible enum also means that previously unseen values may be round-tripped, so not only would I expect `{ "color1": "red", "color2": "red" }` to result in `result.Color1 == result.Color2`, but also `{ "color1": "blue as the sky", "color2": "blue as the sky" }` to result in `result.Color1 == result.Color2` (where that value isn't well known). Without caching, this is not really possible, i.e. those two properties will *not* be referentially equal. But then, imagine the service adds that value to its set of known values (arguably that's exactly what we'd expect in Azure land). Suddenly the values become referentially equal? I'm scared of turning breaking changes into seemingly unrelated behavioral changes.

Reply:
@dsgouda Your default instinct when considering shared mutable state should be to run away screaming. :) There should be a really, really compelling reason to have it, or else by default it's a bad idea.

@olydis The intent is to provide a small optimization for the set of known values. It's more about saving memory than about how equality is implemented. These things should have value equality semantics just like enums, regardless of how the values are constructed. If most of the time that equality check happens to use reference equality under the hood, that's fine. I'm not sure what risk you're seeing here.

Reply:
@brjohnstmsft You're right, I missed the `==` operator definition - think I've spent too much time in JavaScript land and figured that two objects will only compare `==` if they are truly the same. Which is obviously not the case in C#. And there's indeed no requirement in making `Object.ReferenceEquals` "work" since that doesn't work on real `enum`s either due to boxing... thanks for making me think :D

Reply:
so I guess the choice is:
- use caching, so one can get rid of *all* `GetHashCode`/`Equals`/`==` operator definitions
- get rid of the cache, keeping all the operators

Reply:
Why isn't leaving it the way I originally designed it an option? i.e. -- cache known values to save memory and keep the current equality semantics.

Reply:
@olydis Can we please close on this?

Reply:
@fearthecowboy @shahabhijeet are also taking a look into this, they should be posting their comments soon

Reply:
We haven't settled on what to do with the concurrent dictionary. Just to re-raise my question above:

> Why isn't leaving it the way I originally designed it an option? i.e. -- cache known values to save memory and keep the current equality semantics.

Reply:
@fearthecowboy @dsgouda the decision here also affects how we are able to handle "alternative values" (do they roundtrip *and* compare equal to each other?)

Reply:
@olydis I think they should, and the original implementation supported this.